JsonFx/JsonFx.Json.UnitTests/SurrogateTest.cs

```diff
@@ -10,13 +10,31 @@ namespace JsonFx.Json
     [TestFixture()]
     class SurrogateTest
     {
-        	
 		[Test()]
-		public void TestSurrogate()
+		public void TestSurrogates()
 		{
-		    var input = File.ReadAllText("UnitTests/surrogate-test.json");
-		    var deserialized =JsonReader.Deserialize<Dictionary<string,string>>(input);
-            Assert.AreEqual("🔥5             🚒2",deserialized["title"]);
+			var tests = new Dictionary<string,string>(){
+				
+				{"Test UTF-8 heart eyes 1 😍 ",           @"{""inner"":""Test UTF-8 heart eyes 1 \uD83D\uDE0D ""}"},
+				{"Test UTF-8 heart eyes 2 \xd83d\xde0d ", @"{""inner"":""Test UTF-8 heart eyes 2 \uD83D\uDE0D ""}"},
+				{"Test UTF-8 heart eyes 3 \xd83d\xde0d", @"{""inner"":""Test UTF-8 heart eyes 3 \uD83D\uDE0D""}"},
+				{"\xd83d\xde0d Test UTF-8 heart eyes 4", @"{""inner"":""\uD83D\uDE0D Test UTF-8 heart eyes 4""}"},
+				{"\xd83d\xde0d Test UTF-8 heart eyes 5\xd83d\xde0d", @"{""inner"":""\uD83D\uDE0D Test UTF-8 heart eyes 5\uD83D\uDE0D""}"},
+				{"Test UTF-8 g with caron Ǧ ", @"{""inner"":""Test UTF-8 g with caron \u01E6 ""}"},
+				{"Test UTF-8 345f 㑟 ", @"{""inner"":""Test UTF-8 345f \u345F ""}"},
+				{"Test UTF-8 2a7ed � ", @"{""inner"":""Test UTF-8 2a7ed \uFFFD ""}"},
+				{"\xd83d\xdd25             \xd83d\xde92",@"{""inner"":""\uD83D\uDD25             \uD83D\uDE92""}"}
```

Comment:
It would be good for each of these test cases to document the facet which is under test. What for example is the difference between test heart eyes 2, and 3?